ui-admin/src/study/participants/datarepo/DatasetDashboard.tsx

```diff
@@ -0,0 +1,142 @@
+import React, { useEffect, useState } from 'react'
+import { getDatasetListViewPath, StudyEnvContextT } from 'study/StudyEnvironmentRouter'
+import Api, { DatasetDetails, DatasetJobHistory } from 'api/api'
+import LoadingSpinner from 'util/LoadingSpinner'
+import { Store } from 'react-notifications-component'
+import { failureNotification } from 'util/notifications'
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+import { faExternalLink, faArrowLeft } from '@fortawesome/free-solid-svg-icons'
+import { instantToDefaultString } from '../../../util/timeUtils'
+import {
+  ColumnDef,
+  getCoreRowModel,
+  getSortedRowModel,
+  SortingState,
+  useReactTable
+} from '@tanstack/react-table'
+import { basicTableLayout } from '../../../util/tableUtils'
+import { Link, useParams } from 'react-router-dom'
+
+const columns: ColumnDef<DatasetJobHistory>[] = [{
+  id: 'jobType',
+  header: 'Action',
+  accessorKey: 'jobType'
+}, {
+  id: 'status',
+  header: 'Status',
+  accessorKey: 'status'
+}, {
+  id: 'created',
+  header: 'Created',
+  accessorKey: 'createdAt',
+  cell: info => instantToDefaultString(info.getValue() as unknown as number)
+}, {
+  id: 'lastUpdated',
+  header: 'Last Updated',
+  accessorKey: 'lastUpdatedAt',
+  cell: info => instantToDefaultString(info.getValue() as unknown as number)
+}, {
+  id: 'jobDetails',
+  header: 'Job Details',
+  accessorKey: 'tdrJobId',
+  cell: info => <a href={
+    `https://jade.datarepo-dev.broadinstitute.org/activity?expandedJob=${info.getValue()}`} target="_blank"
+  >View job in Terra Data Repo <FontAwesomeIcon icon={faExternalLink}/></a>
+}]
+
+const DatasetDashboard = ({ studyEnvContext }: {studyEnvContext: StudyEnvContextT}) => {
+  const { currentEnvPath } = studyEnvContext
+  const [datasetDetails, setDatasetDetails] = useState<DatasetDetails | null>(null)
+  const [datasetJobHistory, setDatasetJobHistory] = useState<DatasetJobHistory[]>([])
+  const [isLoading, setIsLoading] = useState(true)
+  const [sorting, setSorting] = React.useState<SortingState>([])
+  const datasetName = useParams().datasetName as string
+
+  const table = useReactTable({
+    data: datasetJobHistory,
+    columns,
+    state: {
+      sorting
+    },
+    onSortingChange: setSorting,
+    getCoreRowModel: getCoreRowModel(),
+    getSortedRowModel: getSortedRowModel(),
+    debugTable: true
+  })
+
+  const contentHeaderStyle = {
+    padding: '1em 0 0 1em',
+    borderBottom: '1px solid #f6f6f6'
+  }
+
+  const loadData = async () => {
```

Comment:
Nit: `useEffect` functions like this are usually declared inside `useEffect`. I don't remember all of the details about why. I think technically, since it's declared outside, it should be in the `useEffect` dependency list. However, in reality, this function may be stable across re-renders... which I think in this case is going to depend on whether or not `datasetName` is stable; I imagine it is because it's coming from the query string, but depending on how routing works, that's not universally guaranteed (though it might be in this project). (It's also possible that I missed other dependencies; I only noticed `datasetName` (and `studyEnvContext` which I'm pretty sure is fine because it's a prop).) If the function is in fact stable, then the only reason to put it in `useEffect` might be to make the linter happy... which doesn't seem to be a problem in Juniper, but Terra UI has a linter rule that does an exhaustive dependency check and would complain about this.